payroll-processor-client/src/app/data/state/data-model.ts

```diff
@@ -0,0 +1,9 @@
+export class NameValue {
```

Comment:
I'd recommend using classes when you need to generate derived state:

```typescript
class User {
   firstName = 'test';
   lastName = 'user';

   get fullName() { return `${this.firstName} ${this.lastName}`; }
}
```

If you don't have derived state, an interface is a better option because it's committing to less 'structure' too early.

```typescript
interface NameValue {
  readonly name: string;
  readonly value: number;
}
```